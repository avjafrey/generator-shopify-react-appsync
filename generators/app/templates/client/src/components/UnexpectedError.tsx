import { EmptyState } from "@shopify/polaris";
import * as React from "react";

export function UnexpectedError() {
    return (
        <main role="main">
            <EmptyState
                heading="Unexpected Error"
                action={{ content: "Application Home", url: "/" }}
                image="https://cdn.shopify.com/s/files/1/0757/9955/files/empty-state.svg"
            >
                <p>Well this is embarrasing. Something unexpected happened and we couldn't recover.</p>
            </EmptyState>
        </main>
    );
}
