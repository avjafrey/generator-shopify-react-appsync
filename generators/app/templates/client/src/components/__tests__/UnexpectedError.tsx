import { AppProvider } from "@shopify/polaris";
import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { UnexpectedError } from "../UnexpectedError";

test("Renders correctly", () => {
  const login = (
    <AppProvider>
      <UnexpectedError />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
