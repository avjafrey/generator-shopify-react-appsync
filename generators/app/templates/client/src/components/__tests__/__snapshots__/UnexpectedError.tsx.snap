// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<main
  role="main"
>
  <div
    className="Polaris-EmptyState"
  >
    <div
      className="Polaris-EmptyState__Section"
    >
      <div
        className="Polaris-EmptyState__DetailsContainer"
      >
        <div
          className="Polaris-EmptyState__Details"
        >
          <div
            className="Polaris-TextContainer"
          >
            <p
              className="Polaris-DisplayText Polaris-DisplayText--sizeMedium"
            >
              Unexpected Error
            </p>
            <div
              className="Polaris-EmptyState__Content"
            >
              <p>
                Well this is embarrasing. Something unexpected happened and we couldn't recover.
              </p>
            </div>
          </div>
          <div
            className="Polaris-EmptyState__Actions"
          >
            <div
              className="Polaris-ButtonGroup"
            >
              <div
                className="Polaris-ButtonGroup__Item"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <a
                  className="Polaris-Button Polaris-Button--primary Polaris-Button--sizeLarge"
                  data-polaris-unstyled={true}
                  href="/"
                  onMouseUp={[Function]}
                >
                  <span
                    className="Polaris-Button__Content"
                  >
                    <span>
                      Application Home
                    </span>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Polaris-EmptyState__ImageContainer"
      >
        <img
          alt=""
          className="Polaris-EmptyState__Image"
          role="presentation"
          src="https://cdn.shopify.com/s/files/1/0757/9955/files/empty-state.svg"
        />
      </div>
    </div>
  </div>
</main>
`;
